File: go-version/leetcode/LC2537/count_good.go
package LC2537

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/4/16 上午8:30
 * @FilePath: count_good
 * @Description: 滑动窗口
 */

func countGood(nums []int, k int) int64 {
	cnt := make(map[int]int)
	var ans int64
	left, pairs := 0, 0
	for _, num := range nums {
		pairs += cnt[num]
		cnt[num]++
		for pairs >= k {
			pairs -= cnt[nums[left]] - 1
			cnt[nums[left]]--
			left++
		}
		ans += int64(left)
	}
	return ans
}

File: python-version/LC2537/count_good.py
class Solution:
    def countGood(self, nums: list[int], k: int) -> int:
        ans = 0
        left, pairs = 0, 0
        cnt = {}
        for num in nums:
            pairs += cnt.get(num, 0)
            cnt[num] = cnt.get(num, 0) + 1
            while pairs >= k:
                pairs -= cnt[nums[left]] - 1
                cnt[nums[left]] = cnt[nums[left]] - 1
                left += 1
            ans += left
        return ans
